pkg/build/apiserver/registry/buildlog/rest_test.go

```diff
@@ -130,62 +167,124 @@ func TestWaitForBuild(t *testing.T) {
+		// "status" described on each test-case
+		for _, status := range tt.status {
+			updatedBuild := mockBuild(status, name, version)
+
+			_, err := buildClient.BuildV1().
+				Builds(corev1.NamespaceDefault).
+				Update(informersCtx, updatedBuild, metav1.UpdateOptions{})
+			if err != nil {
+				t.Errorf("Error updating build '%s': '%#v'", name, err)
 			}
-		}()
+		}
+
+		// artificially waiting for informer's cache synchronization
+		if !cache.WaitForCacheSync(
+			informersCtx.Done(),
+			buildInformer.Informer().HasSynced,
+			podInformer.Informer().HasSynced,
+		) {
+			t.Error("Informer's cache is not updated!")
+		}
+
+		// the object namespace is deducted using the informed context, thus creating a custom
+		// context using "default" namespace, the same mocked objects are using
+		ctx := apirequest.NewDefaultContext()
 
 		_, err := storage.Get(ctx, build.Name, &buildapi.BuildLogOptions{})
+		if err != nil {
+			t.Logf("Storage update error: '%#v'", err.Error())
+		}
 		if tt.expectError && err == nil {
 			t.Errorf("%s: Expected an error but got nil from waitFromBuild", tt.name)
 		}
 		if !tt.expectError && err != nil {
-			t.Errorf("%s: Unexpected error from watchBuild: %v", tt.name, err)
+			t.Errorf("%s: Unexpected error from watchBuild: '%v'", tt.name, err)
 		}
+
+		cancel()
 	}
 }
 
 func TestWaitForBuildTimeout(t *testing.T) {
+	informersCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	client := newPodClient()
+	podInformer := fakeCoreV1PodInformer(client, informersCtx.Done())
+
 	build := mockBuild(buildv1.BuildPhasePending, "running", 1)
 	buildClient := buildfakeclient.NewSimpleClientset(build)
+	buildInformer := fakeBuildV1Informer(buildClient, informersCtx.Done())
+
 	ctx := apirequest.NewDefaultContext()
 	storage := REST{
 		BuildClient: buildClient.BuildV1(),
-		PodClient:   newPodClient().CoreV1(),
+		PodClient:   client.CoreV1(),
+		BuildLister: buildInformer.Lister(),
+		PodLister:   podInformer.Lister(),
 		Timeout:     100 * time.Millisecond,
 	}
 
+	// artificially waiting for informer's cache synchronization
+	if !cache.WaitForCacheSync(
+		informersCtx.Done(),
+		buildInformer.Informer().HasSynced,
+		podInformer.Informer().HasSynced,
+	) {
+		t.Error("Informer's cache is not updated!")
+	}
+
 	_, err := storage.Get(ctx, build.Name, &buildapi.BuildLogOptions{})
 	if err == nil || !strings.Contains(err.Error(), "timed out") {
 		t.Errorf("Unexpected error result from waitForBuild: %v\n", err)
 	}
 }
 
-func resourceLocationHelper(buildPhase buildv1.BuildPhase, podPhase string, ctx context.Context, version int) (string, string, string, error) {
+func resourceLocationHelper(ctx context.Context, buildPhase buildv1.BuildPhase, podPhase string, version int) (string, string, string, error) {
+	informersCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	client := newPodClient()
+	podInformer := fakeCoreV1PodInformer(client, informersCtx.Done())
```

Comment:
Assuming using the provided context:
```suggestion
	podInformer := fakeCoreV1PodInformer(client, ctx.Done())
```

Reply:
Please consider [this comment](https://github.com/openshift/openshift-apiserver/pull/206#discussion_r630787588).